lib/esformatter.js

```diff
@@ -41,16 +41,19 @@ exports.unregister = plugins.unregister;
 
 
 function format(str, opts) {
+  _options.set(opts);
   str = plugins.stringBefore(str);
   var ast = rocambole.parse(str);
-  transform(ast, opts);
+  transform(ast, opts, true /*options already set, so skip setting there here*/);
```

Comment:
just opened the `options.js` file and `plugins.js` and saw a couple weird things that can/should be improved:

**options.js:**

 - `plugins.setOptions(_curOpts);` should probably be `plugins.setOptions(_curOpts.plugins);` to match the other `setOptions` calls.

**plugins.js:** 
 
 - the `setOptions` method should probably clean all the registered plugins before loading new ones (imagine a case where user calls `transform` multiple times in a row using different plugins..)
 - extract the logic that require/register the plugins from `setOptions` into another method and call that at the beginning of `format` (eg. `plugins.load(opts && opts.plugins);`)

if you do that we don't need the `skipOptions` flag anymore. the flag was hiding a few design mistakes.